# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/src/AsyncPanZoomController.cpp
# Commit: 3f6178ca6303
# Full Hash: 3f6178ca63039167d512ef89e3e37ba179bc54c1
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2021-04-27 09:55:09
# Regressor Bug: 1707519
# File Overlap Count: 1
# Description:
#   Bug 1707519 - Set the APZC state NOTHING prior to nullify mAnimation. r=botond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113291
# ==============================================================================

diff -r bacefd6c7f00 -r 3f6178ca6303 gfx/layers/apz/src/AsyncPanZoomController.cpp
--- a/gfx/layers/apz/src/AsyncPanZoomController.cpp	Tue Apr 27 01:36:30 2021 +0000
+++ b/gfx/layers/apz/src/AsyncPanZoomController.cpp	Tue Apr 27 03:14:22 2021 +0000
@@ -4406,8 +4406,8 @@
     bool wantsRepaints = mAnimation->WantsRepaints();
     *aOutDeferredTasks = mAnimation->TakeDeferredTasks();
     if (!continueAnimation) {
+      SetState(NOTHING);
       mAnimation = nullptr;
-      SetState(NOTHING);
     }
     // Request a repaint at the end of the animation in case something such as a
     // call to NotifyLayersUpdated was invoked during the animation and Gecko's
